### PR TITLE
Address change in lobbyist structure, fix bug preventing download of all filings

### PR DESCRIPTION
**Pull request by hancush**
Successful run here: https://github.com/datamade/nmid-scrapers/actions/runs/11354821874/job/31582862346

### CODE REVIEW

File: .github/workflows/tmp.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ jobs:
   build:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (hancush) [inline .github/workflows/tmp.yml:16]: Parallelizes building of these files.

File: scrapers/lobbyist/scrape_filings.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def scrape(self, id, version):
             for record in response.json():
                 # Add the lobbyist ID and version to the filing record
                 record["MemberID"] = id
-                return record
+                yield record
```
Comment (hancush) [inline scrapers/lobbyist/scrape_filings.py:44]: The source of pain! 🙀

File: scrapers/lobbyist/cli.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+def scrapelib_opts(f):
+    @click.option("--rpm", default=180, show_default=True)
+    @click.option("--retries", default=3, show_default=True)
+    @click.option("--verify/--no-verify", default=False, show_default=True)
+    @functools.wraps(f)
+    def wrapped_func(*args, **kwargs):
+        return f(*args, **kwargs)
+
+    return wrapped_func
```
Comment (hancush) [inline scrapers/lobbyist/cli.py:14]: This was the main reason for pulling this into a CLI. That way, we can run scrapes faster locally by specifying --rpm=something larger than 3

File: scrapers/lobbyist/scrape_employers.py [hunk 4]
```diff
@@ -59,15 +60,20 @@ def _employers(self):
                 sys.exit()
 
     def scrape(self):
+        seen_employers = deque(maxlen=25)
```
Comment (hancush) [inline scrapers/lobbyist/scrape_employers.py:63]: Added this so the output is deduplicated.

File: scrapers/lobbyist/scrape_lobbyists.py [hunk 5]
```diff
@@ -1,90 +1,103 @@
...
+        "AgentType": "",
+        "PageNo": 1,
+        "PageSize": 1000,
+        "SortOrder": "ASC",
+    }
+
+    page_number = 1
+    page_size = 1000
+    result_count = 1000
+    seen_lobbyists = deque(maxlen=25)
```
Comment (hancush) [inline scrapers/lobbyist/scrape_lobbyists.py:54]: Ditto deduplicated. In addition, our previous approach for scraping lobbyist employers -> scraping lobbyists no longer works????? If you go to the lobbyist tab for an employer with multiple lobbyists and click around the list, you'll notice the lobbyist IDs are not correct and the links don't resolve correctly. So, I updated this script to scrape the lobbyist search, and for each distinct lobbyist, retrieve details. I scraped their employers (clients) separately.

**Review by antidipyramid — APPROVED** [review]

**Pull request merged**